File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/rules/RecordIsAuditable.java
package com.freight.rules;

/** Triage predicate: record is auditable. */
public final class RecordIsAuditable implements TriageRule {

    @Override
    public String name() {
        return "record_is_auditable";
    }

    @Override
    public boolean apply(ProbeRecord record) {
        return new LaneIndexInRange().apply(record) && new SealLengthIsCanonical().apply(record) && !new MassExceedsSoftCap().apply(record);
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/rules/record_is_auditable.cpp
#include "freight/rules.h"

namespace freight {

// Triage predicate: record is auditable.
bool rule_record_is_auditable(const ProbeRecord& record) {
  return rule_lane_index_in_range(record) && rule_seal_length_is_canonical(record) && !rule_mass_exceeds_soft_cap(record);
}

}  // namespace freight
